presto-tests/src/test/java/io/prestosql/tests/TestTpchTableScanRedirection.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.tests;
+
+import io.prestosql.plugin.memory.MemoryPlugin;
+import io.prestosql.testing.AbstractTestQueryFramework;
+import io.prestosql.testing.DistributedQueryRunner;
+import io.prestosql.testing.QueryRunner;
+import io.prestosql.tests.tpch.TpchQueryRunnerBuilder;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+
+public class TestTpchTableScanRedirection
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        DistributedQueryRunner queryRunner = TpchQueryRunnerBuilder.builder()
+                .withTableScanRedirectionCatalog("memory")
+                .withTableScanRedirectionSchema("test")
+                .build();
+        queryRunner.installPlugin(new MemoryPlugin());
+        queryRunner.createCatalog("memory", "memory");
+        // Add another tpch catalog without redirection to aid in loading data into memory connector
+        queryRunner.createCatalog("tpch_data_load", "tpch");
+        return queryRunner;
+    }
+
+    @Test(timeOut = 20_000)
+    public void testTableScanRedirection()
+    {
+        assertQuerySucceeds("CREATE SCHEMA memory.test");
+        assertUpdate("CREATE TABLE memory.test.orders AS SELECT * FROM tpch_data_load.tiny.orders WHERE orderstatus IN ('O', 'P')", 7696L);
+        assertEquals(computeActual("SELECT * FROM tpch.tiny.orders WHERE orderstatus IN ('O', 'F')").getRowCount(), 7333L);
```

Comment:
why difference between `7696L` and `7333L`?

Reply:
orderstatus in orders has possible values `O`, `P` and `F`
I loaded only the rows corresponding to `O` and `P` in redirected table to create different result from querying original table
Then in the test query I selected only the rows corresponding to `O` and `F` in redirected table to test out re-materialisation of filter after redirection to memory onnector while also having different result than if we had queried the original table.

Reply:
> test out re-materialisation of filter after redirection to memory onnector while also having different result than if we had queried the original table.

How do you know it rematerialized (I suppose we only select data for `O`)? Please add a comment.

Reply:
added comments